### ISSUE TITLE
Shapefiles from a Coordinate Reference System are invisible

### ISSUE THREAD

**Issue by montyjohn** [wfletcher/EasyGIS.NET#14]
Hi,
It may be be me doing something wrong but I'm finding that when:

    `sfMap1.UseMercatorProjection = true;`

If I bring in a shp file using a different CRS (BritishNationalGrid in my instance) the shapes are invisible. They are there because I can see the cursor showing data form the shapes (not sure if they are in the right place or not, I'm not sure if the conversions automatic or not).

If I set:

    `sfMap1.UseMercatorProjection = false;`

Then the shp file is perfectly visible.

I've asked the question here but not getting any answers. 
The is more detail in the StackOverflow quesiton.

[StackOverflow Question](https://stackoverflow.com/q/66872166/10079909?sem=2 )

Example 6 behaves the same way.

Thanks for any help.

**Comment by wfletcher** [comment]
Hi,
About 12 months ago I added proper support for different CRS and projections, which is implemented using the native proj6 library.  Before this the only projection that was available was the Mercator Projection and this was only supported if your shapefile was using WGS84 geodetic coordinates (lat/lon). No checks were made and if you set the UseMercatorProjection projection to true it assumed your source shapefile was using geodetic lat/lon coordinates. In your case because your shapefile is using a projected CRS the library is trying to apply a Mercator Projection, but because your coordinates are not with +/- 180 and +/-85 degrees they get projected to invalid coordinates. 

I am planning on removing the UseMercatorProjection property option, but for the moment it is still used by the web edition and Example 6 from the EGIS website (https://easygisdotnet.com/code-examples.html). If you plan on using these at the moment you will need to convert your shapefile CRS to WGS84, which can be done with this utility. https://easygisdotnet.com/downloads/CRSTransformationUtility.zip

Also if you wish to change you map CRS you can do this from the options. 
![image](https://user-images.githubusercontent.com/12992602/113227711-0db10780-92df-11eb-9475-f5c055bd42ee.png)
![image](https://user-images.githubusercontent.com/12992602/113227752-23bec800-92df-11eb-8a7f-25616a04d793.png)

**Comment by montyjohn** [comment]
Thanks for your response.
It makes perfect sense.
I think where I got confused is on reading the FAQ:

"**Does Easy GIS .NET support projections?**
Yes - Easy GIS .NET supports over 5500 Coordinate Reference Systems(CRS) and performs on-the-fly CRS transformations if needed. Projection support was added in version 4.6.0 and is implemented using native proj6 DLLs. See proj.org"

I read this to mean that one can open a SHP file of different CRS (for example British National Grid) and project to whatever projection you are using without first having to convert and save the SHP file to WGS84. Although thinking about it now, I can use your response, to get all that to happen in the background so I guess it is on the fly with a little bit of work and I suspect this is what you meant in the FAQ response.

As a side note, as someone who is fairly new to writing code, C#, GIS and everything I'm playing with here, the reason I opted to use EasyGIS as apposed to the various other C# mapping tools out there is because your example were so simple and useful that it made the library accessible to me. Without them it would have been a non starter for me.

Thanks again.
